CHANGELOG.md

```diff
@@ -3,6 +3,26 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/)
 
+## 2.2.0 - 2022-5-03
+
+### Added
+
+Added unconfigured rules that have been added to rubocop since the last version we used. Set `Enabled` to false for all of them:
+- Lint/RaiseException
```

Comment:
How did you find this list?

Reply:
When I ran Rubocop in Viceroy after updating the rubocop gem, it complained and gave me that list.

Reply:
Oof, did Viceroy fail on those two lint errors???

We should probably try it in VBE as well.

Reply:
Actually, it didn't fail on those. It just complained that they weren't configured prior to configuring them. 

<img width="1447" alt="Screen Shot 2022-05-03 at 3 39 50 PM" src="https://user-images.githubusercontent.com/37208150/166553256-4ced4fab-11bc-48d4-a293-5219fd4bf8ce.png">

Reply:
awesome.
Looks like there's some failures in VBE 😬 
![image](https://user-images.githubusercontent.com/11837546/166554126-29dba3d5-a464-48eb-8a47-8b479bd50bf8.png)

Reply:
Although, there are some failures when I run `rubocop` in Viceroy now:

```shell
orenmagid@C02FK0F8MD6P ruby % rubocop
Inspecting 73 files
..CC.C.CW...................C.....C.....................................C

Offenses:

lib/generators/install/install_generator.rb:7:24: C: Style/ExpandPathArguments: Use expand_path('templates', __dir__) instead of expand_path('../templates', __FILE__).
      source_root File.expand_path('../templates', __FILE__)
                       ^^^^^^^^^^^
lib/generators/model/model_generator.rb:8:24: C: Style/ExpandPathArguments: Use expand_path('templates', __dir__) instead of expand_path('../templates', __FILE__).
      source_root File.expand_path('../templates', __FILE__)
                       ^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:10:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'routes'                    => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:11:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_routes'        => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:12:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_class'         => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:13:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_test'          => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:14:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_fixture'       => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:15:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_factory'       => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:16:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'position_in_serializer'    => 'before',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:17:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'show_foreign_keys'         => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:19:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'show_indexes'              => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:20:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'simple_indexes'            => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:21:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'model_dir'                 => 'lib/viceroy/models',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:22:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'root_dir'                  => '',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:23:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'include_version'           => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:24:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'require'                   => '',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:25:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_tests'             => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:26:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_fixtures'          => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:27:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_factories'         => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:28:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_serializers'       => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:29:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_scaffolds'         => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:30:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_controllers'       => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:31:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_helpers'           => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:32:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'exclude_sti_subclasses'    => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:33:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'ignore_model_sub_dir'      => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:34:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'ignore_columns'            => nil,
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:35:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'ignore_routes'             => nil,
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:36:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'ignore_unknown_models'     => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:37:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'hide_limit_column_types'   => 'boolean',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:39:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'skip_on_db_migrate'        => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:40:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'format_bare'               => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:41:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'format_rdoc'               => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:42:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'format_markdown'           => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:43:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'sort'                      => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:44:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'force'                     => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:45:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'frozen'                    => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:46:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'classified_sort'           => 'true',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:47:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'trace'                     => 'false',
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:48:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'wrapper_open'              => nil,
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:49:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'wrapper_close'             => nil,
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/auto_annotate_models.rake:50:7: C: Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
      'with_comment'              => 'true'
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/migrate_util.rake:6:3: C: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
  return 0 if value.nil?
  ^^^^^^^^^^^^^^^^^^^^^^
lib/viceroy.rb:2:3: W: Lint/NonDeterministicRequireOrder: Sort files before requiring them.
  Dir[File.join(__dir__, path)].each { |f| require f }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/viceroy/rule_processor/compatibility.rb:36:13: C: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
            next unless ExpressionProcessor.apply(parse_rule(scope), current_asset_attributes(action))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/viceroy/rule_processor/compatibility.rb:112:13: C: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
            next if result.relationship_details[relationship]&.failed?
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
script/rails:4:17: C: Style/ExpandPathArguments: Use expand_path('../config/application', __dir__) instead of expand_path('../../config/application', __FILE__).
APP_PATH = File.expand_path('../../config/application', __FILE__)
                ^^^^^^^^^^^
script/rails:5:14: C: Style/ExpandPathArguments: Use expand_path('../config/boot', __dir__) instead of expand_path('../../config/boot', __FILE__).
require File.expand_path('../../config/boot', __FILE__)
             ^^^^^^^^^^^
viceroy.gemspec:1:12: C: Style/ExpandPathArguments: Use expand_path('lib', __dir__) instead of expand_path('../lib', __FILE__).
lib = File.expand_path('../lib', __FILE__)
           ^^^^^^^^^^^
viceroy.gemspec:24:31: C: Style/ExpandPathArguments: Use expand_path(__dir__) instead of expand_path('..', __FILE__).
  spec.files = Dir.chdir(File.expand_path('..', __FILE__)) do
                              ^^^^^^^^^^^

73 files inspected, 49 offenses detected
```

Reply:
Sounds good. Let me know what the BE committee decides and I'm happy to either enable/disable them. Thanks!

Reply:
@PeterLanglois I looked into the failing cops from VBE and Viceroy to see if they were new or changed between 0.63.0 and 0.81.0. While many of them were, some weren't. It seems weird to me that those that weren't added or changed between our current version and the version we're upgrading to with this PR would be failing with this PR if they aren't already failing. Does this seem weird to you?

```yaml
# Old Rubocop dependency: '~> 0.63.0'
# New Rubocop dependency: '~> 0.81.0'

# Reference: https://github.com/rubocop/rubocop/blob/v0.81.0/config/default.yml

# ________________________ FAILING COPS FROM VBE ________________________

# ------------ Added between 0.63.0 and 0.81.0 ------------

Naming/RescuedExceptionsVariableName:
  Description: 'Use consistent rescued exceptions variables naming.'
  Enabled: true
  VersionAdded: '0.67'
  VersionChanged: '0.68'
  PreferredName: e
  
Lint/RedundantCopDisableDirective:
  Description: >-
                 Checks for rubocop:disable comments that can be removed.
                 Note: this cop is not disabled when disabling all cops.
                 It must be explicitly disabled.
  Enabled: true
  VersionAdded: '0.76'

Lint/NonDeterministicRequireOrder:
  Description: 'Always sort arrays returned by Dir.glob when requiring files.'
  Enabled: true
  VersionAdded: '0.78'
  Safe: false

Layout/ArgumentAlignment:
  Description: >-
                 Align the arguments of a method call if they span more
                 than one line.
  StyleGuide: '#no-double-indent'
  Enabled: true
  VersionAdded: '0.68'
  VersionChanged: '0.77'
  # Alignment of arguments in multi-line method calls.
  #
  # The `with_first_argument` style aligns the following lines along the same
  # column as the first parameter.
  #
  #     method_call(a,
  #                 b)
  #
  # The `with_fixed_indentation` style aligns the following lines with one
  # level of indentation relative to the start of the line with the method call.
  #
  #     method_call(a,
  #       b)
  EnforcedStyle: with_first_argument
  SupportedStyles:
    - with_first_argument
    - with_fixed_indentation
  # By default, the indentation width from Layout/IndentationWidth is used
  # But it can be overridden by setting this parameter
  IndentationWidth: ~

Style/FloatDivision:
  Description: 'For performing float division, coerce one side only.'
  StyleGuide: '#float-division'
  Reference: 'https://github.com/rubocop-hq/ruby-style-guide/issues/628'
  Enabled: true
  VersionAdded: '0.72'
  EnforcedStyle: single_coerce
  SupportedStyles:
    - left_coerce
    - right_coerce
    - single_coerce
    - fdiv

# ------------ Changed between 0.63.0 and 0.81.0 ------------

Lint/RedundantStringCoercion:
  Description: 'Checks for Object#to_s usage in string interpolation.'
  StyleGuide: '#no-to-s'
  Enabled: true
  VersionAdded: '0.19'
  VersionChanged: '0.77'

Style/FormatStringToken:
  Description: 'Use a consistent style for format string tokens.'
  Enabled: true
  EnforcedStyle: annotated
  SupportedStyles:
    # Prefer tokens which contain a sprintf like type annotation like
    # `%<name>s`, `%<age>d`, `%<score>f`
    - annotated
    # Prefer simple looking "template" style tokens like `%{name}`, `%{age}`
    - template
    - unannotated
  VersionAdded: '0.49'
  VersionChanged: '0.75'

# ------------ Neither added nor changed between 0.63.0 and 0.81.0 ------------

Style/RedundantReturn:
  Description: "Don't use return where it's not required."
  StyleGuide: '#no-explicit-return'
  Enabled: true
  VersionAdded: '0.10'
  VersionChanged: '0.14'
  # When `true` allows code like `return x, y`.
  AllowMultipleReturnValues: false

Layout/SpaceInsideStringInterpolation:
  Description: 'Checks for padding/surrounding spaces inside string interpolation.'
  StyleGuide: '#string-interpolation'
  Enabled: true
  VersionAdded: '0.49'
  EnforcedStyle: no_space
  SupportedStyles:
    - space
    - no_space

Style/RedundantParentheses:
  Description: "Checks for parentheses that seem not to serve any purpose."
  Enabled: true
  VersionAdded: '0.36'

# ________________________ FAILING COPS FROM VICEROY ________________________

# ------------ Added between 0.63.0 and 0.81.0 ------------

Lint/NonDeterministicRequireOrder:
  Description: 'Always sort arrays returned by Dir.glob when requiring files.'
  Enabled: true
  VersionAdded: '0.78'
  Safe: false

# ------------ Changed between 0.63.0 and 0.81.0 ------------

Layout/HashAlignment:
  Description: >-
    Align the elements of a hash literal if they span more than
    one line.
  Enabled: true
  AllowMultipleStyles: true
  VersionAdded: '0.49'
  VersionChanged: '0.77'
  # Alignment of entries using hash rocket as separator. Valid values are:
  #
  # key - left alignment of keys
  #   'a' => 2
  #   'bb' => 3
  # separator - alignment of hash rockets, keys are right aligned
  #    'a' => 2
  #   'bb' => 3
  # table - left alignment of keys, hash rockets, and values
  #   'a'  => 2
  #   'bb' => 3
  EnforcedHashRocketStyle: key
  SupportedHashRocketStyles:
    - key
    - separator
    - table
  # Alignment of entries using colon as separator. Valid values are:
  #
  # key - left alignment of keys
  #   a: 0
  #   bb: 1
  # separator - alignment of colons, keys are right aligned
  #    a: 0
  #   bb: 1
  # table - left alignment of keys and values
  #   a:  0
  #   bb: 1
  EnforcedColonStyle: key
  SupportedColonStyles:
    - key
    - separator
    - table
  # Select whether hashes that are the last argument in a method call should be
  # inspected? Valid values are:
  #
  # always_inspect - Inspect both implicit and explicit hashes.
  #   Registers an offense for:
  #     function(a: 1,
  #       b: 2)
  #   Registers an offense for:
  #     function({a: 1,
  #       b: 2})
  # always_ignore - Ignore both implicit and explicit hashes.
  #   Accepts:
  #     function(a: 1,
  #       b: 2)
  #   Accepts:
  #     function({a: 1,
  #       b: 2})
  # ignore_implicit - Ignore only implicit hashes.
  #   Accepts:
  #     function(a: 1,
  #       b: 2)
  #   Registers an offense for:
  #     function({a: 1,
  #       b: 2})
  # ignore_explicit - Ignore only explicit hashes.
  #   Accepts:
  #     function({a: 1,
  #       b: 2})
  #   Registers an offense for:
  #     function(a: 1,
  #       b: 2)
  EnforcedLastArgumentHashStyle: always_inspect
  SupportedLastArgumentHashStyles:
    - always_inspect
    - always_ignore
    - ignore_implicit
    - ignore_explicit

# ------------ Neither added nor changed between 0.63.0 and 0.81.0 ------------

Style/ExpandPathArguments:
  Description: "Use `expand_path(__dir__)` instead of `expand_path('..', __FILE__)`."
  Enabled: true
  VersionAdded: '0.53'

Layout/EmptyLineAfterGuardClause:
  Description: 'Add empty line after guard clause.'
  Enabled: true
  VersionAdded: '0.56'
  VersionChanged: '0.59'
```

Reply:
It's weird, but I think that means their documentation is wrong, or they don't document bug fixes as changes.